### PR TITLE
B 2719 refactor card contribution features to add to variant as

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: shared/panels/contribution-sidepanel/contributions-sidepanel.en.json [hunk 1]
```diff
@@ -47,7 +47,12 @@
   },
   "footer": {
     "actions": {
-      "manageInFullPage": "Manage in full page"
+      "manageInFullPage": "Manage in full page",
+      "asContributor": {
+        "deleteComment": "Delete comment",
+        "cancelApplication": "Cancel application",
+        "seePrInGithub": "See PR in GitHub"
```
Comment (haydencleary) [inline shared/panels/contribution-sidepanel/contributions-sidepanel.en.json:54]:
```suggestion
        "seePrOnGithub": "See PR on GitHub"
```

File: shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  if (props.as === ContributionAs.MAINTAINER) {
+    return maintainer;
+  }
+
+  if (props.as === ContributionAs.CONTRIBUTOR) {
+    return contributor;
+  }
+
+  return <div />;
```
Comment (haydencleary) [inline shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx:139]: Can't return `null`?

File: shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+export const useContributionPanelFooterAsMaintainer = ({
+  as,
+  contribution,
+  setIsManageApplicantsModalOpen,
+}: UseContributionPanelFooter) => {
+  const { buttons, endContent } = useContributionActions({ as, contribution });
+  const { isProjectOrganisationMissingPermissions, canCurrentUserUpdatePermissions, setIsGithubPermissionModalOpen } =
+    useGithubPermissionsContext();
+
+  function HandleManageApplicants() {
```
Comment (haydencleary) [inline shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx:24]:
```suggestion
  function handleManageApplicants() {
```

File: shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        translate={{ token: "panels:contribution.footer.actions.manageInFullPage" }}
+      />
+    );
+  }
+
+  if (
+    contribution?.isInProgress() &&
+    contribution.type !== "PULL_REQUEST" &&
+    isProjectOrganisationMissingPermissions(contribution.repo.id) &&
+    canCurrentUserUpdatePermissions(contribution.repo.id)
```
Comment (haydencleary) [inline shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx:52]: I would keep the values of these methods inside variables.

File: shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+export const useContributionPanelFooterAsContributor = ({ contribution }: UseContributionPanelFooter) => {
+  const [shouldDeleteComment, setShouldDeleteComment] = useState(false);
+
+  if (contribution.isArchived()) {
+    return <div />;
+  }
+
+  function onCancelApplication() {
+    console.log(`Should cancel : true & should delete comment : ${shouldDeleteComment}`);
```
Comment (haydencleary) [inline shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx:90]: Method not done yet ?

**Review by haydencleary — CHANGES_REQUESTED** [review]

File: shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx (hunk 2 above)
Comment (alexbeno) [reply shared/panels/contribution-sidepanel/_features/footer/footer.hooks.tsx:139]: should keep empty div the keep the layout

**Review by haydencleary — APPROVED** [review]

**Pull request merged**